### ISSUE TITLE
Introducing GraphQL as an alternative to actions/events

### ISSUE THREAD

**Issue by stevepeak** [microservices/microservice.guide#94]
> Special thanks to @emilebosch for presenting this concept and working on graphql-protocols/graphql-protocols -- he brought this to my attention 👏 

Use GraphQL for actions/events https://github.com/graphql-protocols/graphql-protocols

Concept of how to include GraphQL specifications:
```yml
# microservice.yml
info:
  ...
graphql:
  port: 8080
  file: ./path/to/spec.graphql
environment:
  ...
```

Example GraphQL spec describing an action `postMessage`:
```graphql
# The type you need for your service
type Message {
  message: String
}

# The query interface
interface Postbox {
  messages: [Message]
}

# The mutation interface
interface PostboxMutatations {
  postMessage(message: String): Message
}
```
The client would utilize the standardizations of GraphQL in the following way:
- Query/Mutations are OMG Actions
- Subscriptions are OMG Events

Example of the service being called via GraphQL
```sh
curl -X POST -d 'mutation postMessage(..){...}' https://service/graphql
```

Example of service being called in Storyscript
```story
res = service postMessage message:'hello'
res.message
```

TL;DR

The OMG expresses service action/events as a definition described in the `microservice.yml` but there is another strategy using GraphQL which comes with bonus features (much like OpenAPI/AsyncAPI).

There are some cool advantages for packages like omg-py is that instead of using Flask and REST endpoints you can use GraphQL package (which apparently is in every language) which puts up the server and gives you simple ways to defined endpoint that are strongly typed. Bonus is that the GraphQL tooling then works for microservice actions (ie query/mutations), events (ie subscriptions). 

Another advantage is for the Storyscript runtime being aware of what result information is required. For example, in the query above we use `res.message` and perhaps not another property. Therefore, during the query to the GraphQL service, the runtime could ask specifically for the output it desires to reduce serialization and data waste.
